### PR TITLE
ddl: return warnings when the alter algorithm is COPY, return errors when the algorithm is wrong

### PR DESCRIPTION
**Pull request by winkyao**
### What problem does this PR solve? 

Return warnings only if the specify alter algorithm is COPY, otherwise, return errors when the algorithm is not supported for the correspond DDL types. 

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by winkyao** [comment]
@crazycs520 @zimulala @morgo PTAL

**Comment by morgo** [comment]
LGTM

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -1219,10 +1219,20 @@ func (s *testIntegrationSuite) assertWarningExec(c *C, sql string, expectedWarn
...
+	c.Assert(err, NotNil)
+	c.Assert(expectedWarn.Equal(err), IsTrue, Commentf("error:%v", err))
+}
+
 func (s *testIntegrationSuite) assertAlterWarnExec(c *C, sql string) {
 	s.assertWarningExec(c, sql, ddl.ErrAlterOperationNotSupported)
 }
 
+func (s *testIntegrationSuite) assertAlterErrorExec(c *C, sql string) {
+	s.assertErrorExec(c, sql, ddl.ErrAlterOperationNotSupported)
```
Comment (zimulala) [inline ddl/db_integration_test.go]: Could we replace it with `testErrorCode`?

**Comment by zimulala** [comment]
Do we need to update tidb-test?

**Pull request merged**